CHANGELOG.md

```diff
@@ -6,6 +6,89 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
 
+## [3.0.0] - 2019-03-27
+
+This release comes with a modernized ``ES6``-class structured code base as well
+as some significant local refactoring work regarding how ``Stack`` and ``Memory``
+are organized within the VM execution engine. It also finalizes the module-structuring
+work with separate folders for ``bloom``, ``evm`` and ``state`` related code and
+removes some rarely used parts of the API (``hookedVM``).
+
+All this is to a large extend preparatory work for a ``v4.0.0`` release which will
+follow in the next months with ``TypeScript`` support and more system-wide 
+refactoring work leading to a more modular and expandable VM and providing the
+ground for future ``eWASM`` integration. If you are interested in the release
+process and want to take part in the refactoring discussion see the associated
+issue [#455](https://github.com/ethereumjs/ethereumjs-vm/issues/455).
+
+**VM Refactoring**
+
+- New ``Memory`` class for evm memory manipulation,
+  PR [#442](https://github.com/ethereumjs/ethereumjs-vm/pull/442)
+- Refactored ``Stack`` manipulation in evm,
+  PR [#460](https://github.com/ethereumjs/ethereumjs-vm/pull/460)
+- Dropped ``createHookedVm`` (BREAKING), being made obsolete by the
+  new ``StateManager`` API,
+  PR [#451](https://github.com/ethereumjs/ethereumjs-vm/pull/451)
+- Removed ``fakeBlockchain`` class and associated tests,
+  PR [#466](https://github.com/ethereumjs/ethereumjs-vm/pull/466)
+
+**Modularization**
+
+- Renamed ``vm`` module to ``evm``, move ``precompiles`` to ``evm`` module,
+  PR [#481](https://github.com/ethereumjs/ethereumjs-vm/pull/481)
+- Moved ``stateManager``, ``storageReader`` and ``cache`` to ``state`` module,
+  [#443](https://github.com/ethereumjs/ethereumjs-vm/pull/443)
+- Replaced static VM ``logTable`` with dynamic inline version in ``EXP`` opcode,
+  [#450](https://github.com/ethereumjs/ethereumjs-vm/pull/450)
+  
+**Code Modernization/ES6**
+
+- Converted ``VM`` to ``ES6`` class,
```

Comment:
This PR also drops the `VM.deps` attribute. I don't think anyone was using that, but it might be worth a mention.

Reply:
Have added a note (generally: will push after my comments)